Adding PS Table role def/assignment commands

etcriss opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description

<!-- Please add a brief description of the changes made in this PR. If you have an ongoing or finished cmdlet design, please paste the link below. -->

## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [x] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [x] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [x] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

VeryEarly commented:
Hi @etcriss ,

- How did you get these SDK without modifying README.md? 
- please update src/CosmosDB/CosmosDB/Changelog.md

YanaXu commented:
/azp run

YanaXu commented:
Hi @etcriss,

Please join your github account to Azure org.

- Link your GitHub Account with Microsoft Account on this page: [Open Source Management Portal / Account link](https://repos.opensource.microsoft.com/link)
- Join below organizations on this page: [Open Source Management Portal / Organizations](https://repos.opensource.microsoft.com/orgs)
  - Azure

YanaXu commented:
Hi @etcriss, I've converted this PR to draft. Please check the comments and update the PR when it's ready.

etcriss commented:
@YanaXu, I have addressed the comments and modified the changelog. Can this PR please be moved back to active review and evaluated for merge. CDB needs this to be released as soon as possible to support partners onboarding to the new security framework supporting MI/AAD tokens. Please let me know what if anything I can do to help here.

YanaXu commented:
/azp run

YanaXu commented:
Hi @etcriss, please fix the CI build errors. 
FYI, the code complete date for next release is 01/28.

etcriss commented:
@YanaXu, I have corrected the build error and included working tests with a TestTableRoleCmdlets.json for future test playbacks. Please let me know what if any next steps remain.

etcriss commented:
/azp run

YanaXu commented:
/azp run

etcriss commented:
@YanaXu, can you help me understand what these build failures are? The details/messages are not very helpful in identifying any issues. I've reconfirmed locally that the changes I've made to CosmosDB compile and tests run successfully in VS2022.

[Check failure on line 687 in Build log](https://github.com/Azure/azure-powershell/pull/26838/files#annotation_31667693160) 

@azure-pipelines
azure-pipelines
/ azure-powershell - powershell-core
Build log #L687

Error: The process 'C:\hostedtoolcache\windows\dotnet\dotnet.exe' failed with exit code 1
 [Check failure on line 689 in Build log](https://github.com/Azure/azure-powershell/pull/26838/files#annotation_31667693162) 

@azure-pipelines
azure-pipelines
/ azure-powershell - powershell-core
Build log #L689

Dotnet command failed with non-zero exit code on the following projects : [ '' ]

etcriss commented:
I've updated the packages correctly this time and ensured the tests are still working. Can this please be given another build pass.

etcriss commented:
/azp run

YanaXu commented:
/azp run

etcriss commented:
I've again updated the packages involved trying to solve the generated files problems that the pipelines are encountering. I've now only updated via the readme.md process files involved with the specific new apis this PR is supporting. 

Can I please request another run. Is there anyway I can kick these build pipelines myself? Only having one run a day triggered by another's actions is terribly slow and not allowing me to iterate quickly. 

/azp run

VeryEarly commented on src/CosmosDB/CosmosDB/ChangeLog.md:
```diff
@@ -21,6 +21,9 @@
 * Upgraded Azure.Core to 1.44.1.
 
+## Version 1.16.0
+* Added support for Cosmos DB Table role definition and role assignment related cmdlets.
```
please move line 25 to 22 under "## Upcoming Release" and delete line 24

VeryEarly commented:
/azp run

VeryEarly commented:
/azp run

VeryEarly commented on src/Accounts/Accounts/Utilities/CommandMappings.json:
```diff
@@ -1611,6 +1611,8 @@
       "Get-AzCosmosDBTableBackupInformation": {},
       "Get-AzCosmosDBTableRestorableResource": {},
       "Get-AzCosmosDBTableRestorableTable": {},
+      "Get-AzCosmosDBTableRoleAssignment": {},
```
please revert changes in this file

VeryEarly commented:
/azp run

etcriss commented:
Thank you very much. @VeryEarly, @YanaXu, according to this documentation that I have been following, https://eng.ms/docs/cloud-ai-platform/azure-data/azure-data-azure-databases/cosmos-db/azure-cosmos-db/azure-cosmos-db-team-docs/design/manageability/rp-client-ecosystem, the next step is to request an out of band deployment.

What are the steps to doing this and how can I get that process started?

etcriss commented:
> Thank you very much. @VeryEarly, @YanaXu, according to this documentation that I have been following, https://eng.ms/docs/cloud-ai-platform/azure-data/azure-data-azure-databases/cosmos-db/azure-cosmos-db/azure-cosmos-db-team-docs/design/manageability/rp-client-ecosystem, the next step is to request an out of band deployment.
> 
> What are the steps to doing this and how can I get that process started?

Ping @VeryEarly, @YanaXu, what would be the next steps to requesting a deployment for this work?

NoriZC commented:
Hi @etcriss, this change will be released with our regular release on March 4th by default.
